src/TBuffer.cpp

```diff
@@ -4149,3 +4182,42 @@ QString TBuffer::bufferToHtml( QPoint P1, QPoint P2, bool allowedTimestamps, int
 
     return s;
 }
+
+const QList<QString> TBuffer::getComputerEncodingNames() {
+    QList<QString> encodings;
+    for (auto encoding: csmEncodingTable.keys()) {
+        encodings << encoding;
+    }
+
+    return encodings;
+}
+
+const QList<QString> TBuffer::getFriendlyEncodingNames() {
+    QList<QString> encodings;
```

Comment:
Compared to the previous method (`getgetComputerEncodingNames`) I agree that you will likely have to iterate through the `csmEncodingTable.values()` because I cannot see any short-cut provided by Qt to get just all the `first` or `second` out of a "container" data-type with a `QPair<T1,T2>` value type...